src/docfx/lib/markdown/ResolveXrefExtension.cs

```diff
@@ -21,24 +23,55 @@ internal static class ResolveXrefExtension
              {
                  document.Replace(node =>
                  {
+                     var file = (Document)InclusionContext.File;
                      if (node is XrefInline xref)
                      {
                          var (_, href, display, _) = resolveXref(xref.Href, xref);
                          if (href is null)
                          {
                              var raw = new SourceInfo<string>(xref.GetAttributes().Properties.First(p => p.Key == "data-raw-source").Value, node.ToSourceInfo());
                              var error = raw.Value.StartsWith("@")
-                                 ? Errors.AtUidNotFound((Document)InclusionContext.File, xref.Href, raw)
-                                 : Errors.UidNotFound((Document)InclusionContext.File, xref.Href, raw);
+                                 ? Errors.AtUidNotFound(file, xref.Href, raw)
+                                 : Errors.UidNotFound(file, xref.Href, raw);
 
                              MarkdownUtility.LogError(error);
                              return new LiteralInline(raw);
                          }
                          return new LinkInline(href, null).AppendChild(new LiteralInline(display));
                      }
+                     else if (node is HtmlBlock block)
+                     {
+                         var (errors, result) = ResolveXref(block.Lines.ToString(), block.ToSourceInfo().Line, block);
+                         LogErrors(errors, file, block.Lines.ToString());
+                         block.Lines = new StringLineGroup(result);
+                     }
+                     else if (node is HtmlInline inline)
+                     {
+                         var (errors, result) = ResolveXref(inline.Tag, inline.ToSourceInfo().Line, inline);
+                         LogErrors(errors, file, inline.Tag);
+                         inline.Tag = result;
+                     }
                      return node;
                  });
              });
+
+            void LogErrors(List<(string uid, int line, int column, string errorCode)> errors, Document file, string html)
+            {
+                foreach (var (uid, line, column, errorCode) in errors)
```

Comment:
why not make `TransformXref` return a `List<Error>` to avoid doing the transformation here?

Reply:
Because then we need to send in File as input for `TransformXref`, which would cause problems for the tests sending in null file

Reply:
How about modifying `Errors.UidNotFound` to take `SourceInfo` over `Document` then `TransformXref` could depend on a `string file`?

Reply:
sure, sounds like a plan